ufl/tensors.py

```diff
@@ -120,6 +126,14 @@ def evaluate(self, x, mapping, component, index_values, derivatives=()):
         else:
             return a.evaluate(x, mapping, component, index_values)
 
+    def _simplify_indexed(self, multiindex):
+        """Return a simplified Expr used in the constructor of Indexed(self, multiindex)."""
+        k = multiindex[0]
+        if isinstance(k, FixedIndex):
+            sub = self.ufl_operands[int(k)]
+            return sub if len(multiindex) == 1 else Indexed(sub, MultiIndex(multiindex[1:]))
```

Comment:
```suggestion
            return Indexed(sub, MultiIndex(multiindex[1:]))
```
Indexed.__new__() will handle this.